src/Domain/SubscriptionRepository.php

```diff
@@ -11,6 +11,11 @@
  */
 interface SubscriptionRepository {
 
+	/**
+	 * @param Subscription $subscription
+	 * @return mixed
+	 * @throws RepositoryException
```

Comment:
This is certainly an improvement. However I think it's odd to have a single RepositoryException. This prevents any domain specific information from being added. Example: https://github.com/wmde/WikibaseDatabase/blob/master/src/QueryInterface.php#L18

I'd have not gone that far right off the bat, though still have added a single Exception per repository (or more generally speaking, per service).